packages/host/src/components/Login/hooks.ts

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import create from 'zustand';
+
+import { AppRoute } from '@openmsupply-client/config';
+import {
+  Store,
+  useHostContext,
+  useLocalStorage,
+  useLocation,
+  useNavigate,
+} from '@openmsupply-client/common';
+import { useAuthToken } from './api';
+
+interface LoginForm {
+  isLoggingIn: boolean;
+  password: string;
+  store?: Store;
+  username: string;
+  setIsLoggingIn: (isLoggingIn: boolean) => void;
+  setPassword: (password: string) => void;
+  setStore: (store?: Store) => void;
+  setUsername: (username: string) => void;
+}
+export const useLoginFormState = create<LoginForm>(set => ({
+  isLoggingIn: false,
+  password: '',
+  store: undefined,
+  username: '',
+  setIsLoggingIn: (isLoggingIn: boolean) =>
+    set(state => ({ ...state, isLoggingIn })),
+  setPassword: (password: string) => set(state => ({ ...state, password })),
+  setStore: (store?: Store) => set(state => ({ ...state, store })),
+  setUsername: (username: string) => set(state => ({ ...state, username })),
+}));
```

Comment:
I thought it would be neater than having four `useState`s, and a little more fun. I'm concerned that zustand is only used for global state elsewhere, so maybe it's confusing to have it here, for local state?

I started with the `react-form`.. but that was more overhead than I wanted and not fully working.
Will have a look at changing

Reply:
I think you're right and don't think it needs to change. I actually had been thinking that replacing any use of context for zustand might be better .. it's a lot easier to mess up context than zustand..!